src/renderer/pages/player-page.js

```diff
@@ -54,14 +54,10 @@ function renderMedia (state) {
       mediaElement.pause()
     } else if (!state.playing.isPaused && mediaElement.paused) {
       mediaElement.play()
-        .then(() => {
-          dispatch('mediaSuccess')
-        })
-        .catch((err) => {
-          if (err.name === 'NotSupportedError') {
-            dispatch('mediaError', 'Codec unsupported')
-          }
-        })
+        .then(
+          () => dispatch('mediaSuccess'),
+          () => dispatch('mediaError', 'Codec unsupported')
```

Comment:
Maybe?
```js
        .then(
          () => dispatch('mediaSuccess'),
          err => dispatch('mediaError', err.name === 'NotSupportedError' ? 'Codec unsupported' : `${err.name}: ${err.message}`)
        )
```
I put it on a branch: https://github.com/webtorrent/webtorrent-desktop/tree/issue-1711-improve-error-handling

For some reason I would send PR to your clone, I think select your fork as the base. Maybe a bug in GitHub, to many forks of this repo.

Reply:
I think that we should dispatch `mediaError` when error type is `NotSupportedError` (original code) or when any error case (like the latest code change).

Given that the possible [exceptions](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/play#Exceptions) are `NotSupportedError` or `NotAllowedError` I wouldn't show this last one to the final user 😅

Reply:
> I wouldn't show this last one to the final user

why not?

Reply:
It also says _other exceptions may be reported,_. I suppose there are not very likely, But it may be good user gets an error if something odd happens.

Reply:
With the current code, in the unlikely event of an error named `NotAllowedError`, or any other exception, it will prompt the user with 'Codec unsupported' right?

Reply:
> With the current code, in the unlikely event of an error named NotAllowedError, or any other exception, it will prompt the user with 'Codec unsupported' right?

Yes that is right 😅 Although it is not 100% correct for the final user, the action should be the same: **open an external player to be able to play the file**, otherwise it would see the message `The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.` which I found a bit confusing for the user as it doesn't give any "call to action" to solve the problem.

On the other hand, it is true that we should log the proper error code, so we can debug future problems easily 😉

What do you think @Borewit ?